BB article 3: Vector Search

prrao87 opened the pull request:
Hi @svonava and @ClaireSuperlinked, I've added the draft for the vector search article. I've tried to reduce verbosity where possible, and have covered the general outline we discussed. Let me know your feedback!

To do: 
- [ ] Need the help of @AruneshSingh to clean up the images to match with the overall website style (the ones I have are largely sketches and aren't meant to be final)
- [ ] Update or replace current cover image if required

AruneshSingh commented:
Hey @prrao87 , thanks for the amazing work! I am working on the images and will update here shortly. I am assuming you made these on excalidraw? Would it be possible to share the excalidraw file with me (arunesh@superlinked.com), would make editing the images a little easier for me.

prrao87 commented:
Great! Just sent those over to you 👍🏼

svonava commented on docs/building_blocks/vector_search/readme.md:
```diff
@@ -1,42 +1,157 @@
-2. or you can give an explanation on how it should be and we will help you create one. Please tag arunesh@superlinked.com or @AruneshSingh (GitHub) in this case. -->
+![](../../assets/building_blocks/vector_search/vector-search-cover-sketch.png)
+
+## The exponential growth of data
```
You cover the unstructured data aspect very well, but you focus on it too much - I'd like a more balanced view, because the vectorization of structured data is equally as important.

Couple examples:
1. Node2Vec and GraphSAGE help us embedd graph data / entity relationship structures, which means that the promiximity of the resulting vectors represents closeness in the original graph.
2. https://huggingface.co/blog/autoformer and similar are transformer-based models for timeseries prediction and they perform at SOTA level - which means that the embeddings you can extract from them are / should be a powerful way to describe and compare timeseries, for example for annomaly detection.
3. There is a bunch of work on tabular data embedding, for example https://research.yandex.com/blog/embeddings-for-numerical-features-in-tabular-deep-learning but I'm sure one could find more

prrao87 replied:
Makes sense! Will look into use cases that are structured.

prrao87 commented:
@svonava I've made the updates:

- [x] Divided the headers into sections for structured and unstructured data
- [x] Incorporated the prior you suggested into these sections, as well as other work that informed them
- [x] Rephrased the entire article's viewpoint to account for both structured and unstructured data
- [x] Added a section about fine-tuning alongside the re-ranking section to provide more details of downstream considerations

I didn't update any images so whatever updates there are on @AruneshSingh's end can be incorporated into the main repo once this is merged.

Hope this is better in terms of balance!

svonava commented:
@prrao87 yep this looks good on the overal content front now

@robertdhayanturner this is ready for your eyes - @AruneshSingh @ClaireSuperlinked do we want robert to apply his changes on top of this branch, or we merge and then robert edits?

robertdhayanturner commented:
@AruneshSingh Can we nail down the procedure for my edits and comments on this article before I proceed? cc @svonava

prrao87 commented:
@robertdhayanturner I think it makes sense to edit this PR to point to a new branch `bb-vector-search` on the superlinked repo, merge it there, and you continue making edits to that branch on the superlinked repo, and merge it to `main` via a separate PR once finished? Going forward, I'll chat with @svonava about adding me as a collaborator with push privileges on the superlinked repo.

AruneshSingh commented:
@prrao87 Can you update this PR to merge into review-content branch instead of main. That is where we'll merge all the content moving on.

AruneshSingh commented:
I have updated the merging branch to be `review-content` where @robertdhayanturner can take this forward. Merging now.

daniel-superlinked commented on docs/building_blocks/vector_search/readme.md:
````diff
@@ -1,42 +1,186 @@
+In computer science, a vector is a one-dimensional array of numbers (typically floating point numbers).
+
+```py
+example_4d_vector = [0.1234, 0.3035, 0.9876]
````
Is this a typo? 4d vs 3 numbers in list

daniel-superlinked commented on docs/building_blocks/vector_search/readme.md:
```diff
@@ -1,42 +1,186 @@
+
+## Vector search for structured data
+
+A common misconception of vector search is that it's applicable only to unstructured data. This is not true! Vector embeddings have *always* been at the heart of deep learning models, and this section will highlight how structured data can be leveraged to gain new insights into existing data.
```
misconception *about* vector search?

daniel-superlinked replied:
"structured data can be leveraged to gain new insights into existing data" needs clarification, perhaps:

"how vector search can be used to gain insights into structured data"
